esp/platform/espp.cpp

```diff
@@ -298,14 +329,38 @@ void openEspLogFile(IPropertyTree* envpt, IPropertyTree* procpt)
         lf->setName("esp_main");//override default filename
         lf->setAliasName("esp");
         lf->setMaxLogFileSize(maxLogFileSize);
-        lf->beginLogging();
+
+        //
+        // If log level defined in legacy config, honor it
+        int loglev = -1;
+        if(procpt->hasProp("@logLevel"))
+            loglev = procpt->getPropInt("@logLevel", LogMin);
+
+        if (loglev > -1)
+            lf->setMaxDetail(mapLogLevelToThreshold((unsigned)loglev)); //jlog range set to 1-100
+        //else
+        //handlers are set to jlog.cpp #define DefaultDetail
+
+        fileHandler = lf->beginLogging();
     }
+
+    //only here to manipulate default log level on stderr stream for smoketest purposes
+    Owned<ILogMsgFilter> filter = getCategoryLogMsgFilter(MSGAUD_all, MSGCLS_all, mapLogLevelToThreshold(LogMin), true);
+    queryLogMsgManager()->changeMonitorFilter(queryStderrLogMsgHandler(), filter);
 #else
     setupContainerizedLogMsgHandler();
+
+    //
+    // ESP Log Update
+    // Add custom log monitor to handle elevated context based logging
+    CEspForceLogOutputFilter *filter = new CEspForceLogOutputFilter();
```

Comment:
elevated context base logging is not a very common feature as far as I know, if attaching this filter to the logmsghandler adds any kind of overhead, let's delay loading this as late as possible.